app/controllers/devise_token_auth/concerns/resource_finder.rb

```diff
@@ -20,21 +20,20 @@ def get_case_insensitive_field_from_resource_params(field)
   end
 
   def find_resource(field, value)
-    # fix for mysql default case insensitivity
-    q = "#{field.to_s} = ? AND provider='#{provider.to_s}'"
-    if ActiveRecord::Base.connection.adapter_name.downcase.starts_with? 'mysql'
-      q = 'BINARY ' + q
-    end
-
-    @resource = resource_class.where(q, value).first
+    @resource = if resource_class.try(:connection_config).try(:[], :adapter).try(:include?, 'mysql')
+                  # fix for mysql default case insensitivity
+                  resource_class.where("BINARY #{field} = ? AND provider= ?", value, provider).first
+                else
+                  resource_class.where(field => value, 'provider' => provider).first
+                end
   end
 
   def resource_class(m = nil)
-    if m
-      mapping = Devise.mappings[m]
-    else
-      mapping = Devise.mappings[resource_name] || Devise.mappings.values.first
-    end
```

Comment:
you can do `end.to` instead of aligning a lot of spaces with thew new variable, also, what does `to` ?

Reply:
as you want, I just wanted the code cleaner, but your point makes sense